src/themes/components.tsx

```diff
@@ -0,0 +1,188 @@
+import { Components } from "@mui/material";
+
+import { palette } from "./palette";
+import { typography } from "./typography";
+
+export const components: Components = {
+  MuiButtonBase: {
+    defaultProps: {
+      disableRipple: true,
+    },
+    styleOverrides: {
+      // @ts-expect-error - MUI types are not up to date
+      root: {
+        textTransform: "none !important",
+
+        "& .MuiButton-startIcon": {
+          marginRight: 4,
+        },
+
+        "& .MuiButton-endIcon": {
+          marginLeft: 4,
+        },
+      },
+    },
+  },
+  MuiButton: {
+    defaultProps: {
+      variant: "contained",
+      color: "cobalt",
+      size: "md",
+    },
+    styleOverrides: {
+      root: ({ ownerState }) => {
+        const size = ownerState.size ?? "md";
+        const sizeStyles = (() => {
+          switch (size) {
+            case "sm":
+              return { padding: "10px 16px", ...typography.h6 };
+            case "md":
+              return { padding: "10px 20px", ...typography.h5 };
+            case "lg":
+              return { padding: "10px 20px", ...typography.h4 };
+            case "xl":
+              return { padding: "10px 20px", ...typography.h3 };
+            default:
+              const _exhaustiveCheck: never = size;
+              return _exhaustiveCheck;
+          }
+        })();
+
+        const variant = ownerState.variant ?? "contained";
+        const color =
+          ownerState.color === "inherit"
+            ? "inherit"
+            : palette[ownerState.color ?? "cobalt"];
+
+        const variantColorStyles = (() => {
+          if (!color) {
+            return {};
+          }
+
+          if (color === "inherit") {
+            return {
+              color: "inherit",
+            };
+          }
+
+          switch (variant) {
+            case "contained":
+              return {
+                backgroundColor: color[400],
+                color: "#fff",
+                "&:hover": {
+                  backgroundColor: color[600],
+                },
+              };
+            case "outlined":
+              return {
+                color: color[500],
+                borderColor: color[500],
+                "&:hover": {
+                  borderColor: color[500],
+                  backgroundColor: color[50],
+                },
+              };
+            case "text":
+              return {
+                color: color[700],
+                "&:hover": {
+                  backgroundColor: "transparent",
+                  color: color[900],
+                },
+              };
```

Comment:
Nit: I would have thought that we would not need to manually do the case, that it would be done via other keys in the styleOverrides ?

I guess at some point, this will become part directly of the swiss-federal-ci, so that this code can be removed ?